service/resource/endpoint/delete.go

```diff
@@ -1,11 +1,68 @@
 package endpoint
 
-import "context"
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
 
 func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
-	return nil, nil
+	currentEndpoint, err := toEndpoint(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	desiredEndpoint, err := toEndpoint(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	deleteState := Endpoint{
+		ServiceName:      currentEndpoint.ServiceName,
+		ServiceNamespace: currentEndpoint.ServiceNamespace,
+	}
```

Comment:
I think it could be clearer to just just `map[string]struct{}` and create slice of keys just before returning. Then you can get rid of `containsIP` guards.

Reply:
I added an issue for this #10, I'll clean this once we are certain that the current state works.